docs/design/core/event-streaming/alternate-syntax/README.md

```diff
@@ -0,0 +1,165 @@
+     */
+     
+    Stream<{RESPONSE_EVENT_TYPE}> blockingResponseEventStream();
+    Stream<{OPERATION}Response> blockingResponseStream();
+
+    @Override
+    default void close() {
+        completionFuture().cancel(false);
+    }
+}
+```
+
+This type enables customers to use the operation in either a
+reactive-streams or a Java-8 usage pattern, depending on how they care
+to manage their threads and back-pressure.
+
+It's worth noting that every method on `Running{OPERATION}` is still
+non-blocking and will never throw exceptions directly. Any method that
+returns a type that itself contains blocking methods is prefixed with
+`blocking`, e.g. `Stream<{RESPONSE_EVENT_TYPE}> 
+blockingResponseEventStream()`.
+
+**Example 1: Transcribe's `startStreamTranscription` with Reactive
+Streams**
+
+```Java
+try (TranscribeStreamingAsyncClient client = TranscribeStreamingAsyncClient.create();
+     // Create the connection to transcribe and send the initial request message
+     RunningStartStreamTranscription transcription =
+            client.startStreamTranscription(r -> r.languageCode(LanguageCode.EN_US)
+                                                  .mediaEncoding(MediaEncoding.PCM)
+                                                  .mediaSampleRateHertz(16_000))) {
+
+    // Use RxJava to create the audio stream to be transcribed
+    Publisher<AudioStream> audioPublisher =
+            Bytes.from(audioFile)
+                 .map(SdkBytes::fromByteArray)
+                 .map(bytes -> AudioEvent.builder().audioChunk(bytes).build())
+                 .cast(AudioStream.class);
+
+    // Begin sending the audio data to transcribe, asynchronously
+    transcription.writeAll(audioPublisher);
+
+    // Get a publisher for the transcription
+    Publisher<TranscriptResultStream> transcriptionPublisher = transcription.responseEventPublisher();
+
+    // Use RxJava to log the transcription
+    Flowable.fromPublisher(transcriptionPublisher)
+            .filter(e -> e instanceof TranscriptEvent)
+            .cast(TranscriptEvent.class)
+            .forEach(e -> System.out.println(e.transcript().results()));
+
+    // Wait for the operation to complete
+    transcription.completionFuture().join();
+}
+```
+
+**Example 2: Transcribe's `startStreamTranscription` without Reactive
+Streams**
+
+```Java
+try (TranscribeStreamingAsyncClient client = TranscribeStreamingAsyncClient.create();
+    // Create the connection to transcribe and send the initial request message
+    RunningStartStreamTranscription transcription =
+            client.startStreamTranscription(r -> r.languageCode(LanguageCode.EN_US)
+                                                  .mediaEncoding(MediaEncoding.PCM)
+                                                  .mediaSampleRateHertz(16_000))) {
+    // Asynchronously log response transcription events, as we receive them
+    transcription.readAll(TranscriptEvent.class, e -> System.out.println(e.transcript().results()));
```

Comment:
Can we see if we can drop the future compatibility here and drop the need to cast?

Reply:
That'd be worth talking about for Transcribe and Kinesis. 

It would need to be modeled, since it's something the service needs to opt into, and probably something that should be validated in coralcop - that they can't add anything new.

Let's save that discussion for when we get to actually modeling it?